src/_nebari/upgrade.py

```diff
@@ -1297,10 +1298,16 @@ def _version_specific_upgrade(
 
             urllib3.disable_warnings()
 
+            keycloak_username = os.environ.get("KEYCLOAK_ADMIN_USERNAME", "root")
```

Comment:
I am including a first layer check for the password, in case the user changed it